pkg/telemetry/statsworker.go

```diff
@@ -163,17 +237,111 @@ func coalesce(stats []*livekit.AnalyticsStat) *livekit.AnalyticsStat {
 		}
 	}
 
-	streams := make([]*livekit.AnalyticsStream, 0, len(analyticsStreams))
+	curStats := Stat{Score: score, VideoLayers: make(map[int32]*livekit.AnalyticsVideoLayer)}
+	// find aggregates across streams
 	for ssrc, analyticsStream := range analyticsStreams {
-		stream := proto.Clone(analyticsStream).(*livekit.AnalyticsStream)
-		stream.Rtt = maxRTT[ssrc]
-		stream.Jitter = maxJitter[ssrc]
+		rtt := maxRTT[ssrc]
+		if rtt > curStats.Rtt {
+			curStats.Rtt = rtt
+		}
+
+		jitter := maxJitter[ssrc]
+		if jitter > curStats.Jitter {
+			curStats.Jitter = jitter
+		}
+
+		curStats.TotalFrames += analyticsStream.TotalFrames
+		curStats.TotalPrimaryPackets += analyticsStream.TotalPrimaryPackets
+		curStats.TotalPrimaryBytes += analyticsStream.TotalPrimaryBytes
+		curStats.TotalRetransmitPackets += analyticsStream.TotalRetransmitPackets
+		curStats.TotalRetransmitBytes += analyticsStream.TotalRetransmitBytes
+		curStats.TotalPaddingPackets += analyticsStream.TotalPaddingPackets
+		curStats.TotalPaddingBytes += analyticsStream.TotalPaddingBytes
+		curStats.TotalPacketsLost += analyticsStream.TotalPacketsLost
+		curStats.TotalFrames += analyticsStream.TotalFrames
+		curStats.TotalNacks += analyticsStream.TotalNacks
+		curStats.TotalPlis += analyticsStream.TotalPlis
+		curStats.TotalFirs += analyticsStream.TotalFirs
+		// add/update new video VideoLayers data to current and sum up video layer bytes/packets
+		for _, videoLayer := range analyticsStream.VideoLayers {
+			curStats.VideoLayers[videoLayer.Layer] = proto.Clone(videoLayer).(*livekit.AnalyticsVideoLayer)
+			curStats.TotalPackets += videoLayer.TotalPackets
+			curStats.TotalBytes += videoLayer.TotalBytes
+		}
+
+	}
+
+	// update currentStats
+	stats.curStats = &curStats
+
+	return
+}
+
+// find delta between  curStats and prevStats and prepare proto payload
+func (stats *Stats) computeDeltaStats() *livekit.AnalyticsStat {
 
-		streams = append(streams, stream)
+	if stats.curStats == nil {
+		return nil
+	}
+
+	// Stats in both queue/prev contain consolidated single deltaStats
+	cur := stats.curStats
+
+	var maxLayer int32
+	var maxTotalBytes uint64
+	//create a map of VideoLayers - to pick max/best layer wrt current and prev
+	curLayers := make(map[int32]*livekit.AnalyticsVideoLayer)
+	for _, layer := range cur.VideoLayers {
+		curLayers[layer.Layer] = layer
+		// identify layer which sent max data - as VideoLayers can change in current interval
```

Comment:
Should this be data based or layer based? I guess we can think of this as the most active layer.

Reply:
since this interval based, the layer can have switched between start/end of the interval. With bytes, the idea is to identify the current used max layer for the next round